physics/lsm_ruc.F90

```diff
@@ -995,14 +1010,14 @@ subroutine lsm_ruc_run                                            & ! inputs
         solnet_lnd(i,j) = dswsfc(i)*(1.-alb_lnd(i,j)) !..net sw rad flx (dn-up) at sfc in w/m2
 
         cmc(i,j) = canopy(i)            !  [mm] 
-        soilt_lnd(i,j) = tsurf_lnd(i)            ! clu_q2m_iter
+        soilt_lnd(i,j) = tsurf_lnd(i)   
         ! sanity check for snow temperature tsnow
-        if (tsnow_lnd(i) > 0. .and. tsnow_lnd(i) < 273.15) then
+        if (tsnow_lnd(i) > 200. .and. tsnow_lnd(i) < 273.15) then
           soilt1_lnd(i,j) = tsnow_lnd(i)
         else
           soilt1_lnd(i,j) = tsurf_lnd(i)
         endif
-        tsnav_lnd(i,j) = 0.5*(soilt_lnd(i,j) + soilt1_lnd(i,j)) - 273.15
+        tsnav_lnd(i,j) = min(0.,0.5*(soilt_lnd(i,j) + soilt1_lnd(i,j)) - 273.15)
```

Comment:
Replace **min(0.** with **min(0._kind_phys**

Reply:
@dustinswales and also 0.5 to 0.5_kind_phys?

Reply:
It's best if all constants had the _kind_phys suffix. This way there is no ambiguity about the precision, making the code more portable. It's possible to lose precision w/o notification from the compiler, so it's best to be explicit. https://www.fortran90.org/src/best-practices.html